compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -901,4 +901,31 @@ object messages {
     val msg = hl"trying to define package with same name as `$existing`"
     val explanation = ""
   }
+
+  case class DangelingThisInPath()(implicit ctx: Context) extends Message(34) {
+    val kind = "Syntax"
+    val msg = hl"""Expected an additional member selection after the keyword ${"this"}"""
+
+    val importCode =
+      """import MyClass.this.member
```

Comment:
Maybe use `OuterClass` instead of `MyClass`? And even add `class OuterClass { class InnerClass { ... } }` as context to show the point of `MyClass.this`? I ask because it took me a (small) while to get the point `MyClass.this` (I'll concede I might also be rusty).